### ISSUE TITLE
bundle.t hangs on Windows

### ISSUE THREAD

**Issue by chorny** [jhthorsen/json-validator#102]
Strawberry perl 5.26.0 32-bit + 64-bit int.

```
# [1] replace=1
ok 1 - [1] replace=1
# [1] replace=0
ok 2 - [1] name still in definitions
ok 3 - [1] added to definitions
ok 4 - [1] new age ref
ok 5 - [1] same name ref
ok 6 - [1] age $ref point to /definitions/b_json__definitions_age
ok 7 - [1] name $ref point to /definitions/name
# [2] replace=1
ok 8 - [2] replace=1
# [2] replace=0
ok 9 - [2] name still in definitions
ok 10 - [2] added to definitions
ok 11 - [2] new age ref
ok 12 - [2] same name ref
ok 13 - [2] age $ref point to /definitions/b_json__definitions_age
ok 14 - [2] name $ref point to /definitions/name
# [3] replace=1
ok 15 - [3] replace=1
# [3] replace=0
ok 16 - [3] name still in definitions
ok 17 - [3] added to definitions
ok 18 - [3] new age ref
ok 19 - [3] same name ref
ok 20 - [3] age $ref point to /definitions/b_json__definitions_age
ok 21 - [3] name $ref point to /definitions/name
ok 22 - get /name/$ref
ok 23 - get /name/type
ok 24 - get /name/$ref
ok 25 - schema get /name/type
ok 26 - schema get /name/$ref
ok 27 - no dup definitions
Deep recursion on subroutine "JSON::Validator::_resolve" at C:\Strawberry260\cpan\build\JSON-Validator-2.05-0\blib\lib/JSON/Validator.pm line 432.
Deep recursion on subroutine "JSON::Validator::_resolve_ref" at C:\Strawberry260\cpan\build\JSON-Validator-2.05-0\blib\lib/JSON/Validator.pm line 407.
Deep recursion on subroutine "JSON::Validator::_resolve_ref" at C:\Strawberry260\cpan\build\JSON-Validator-2.05-0\blib\lib/JSON/Validator.pm line 407.
Deep recursion on subroutine "JSON::Validator::_resolve_ref" at C:\Strawberry260\cpan\build\JSON-Validator-2.05-0\blib\lib/JSON/Validator.pm line 407.
(hangs)
```

Also see this cpantesters report: http://www.cpantesters.org/cpan/report/5c245523-6bf6-1014-8ac2-ceba6fcb7d32

**Comment by vejbaek** [comment]
I have had the same problem today. 
A bit of poking around suggests that the issue stems from Mojo::File where path on Windows sometimes returns paths with backslash instead of a slash and Mojo::File uses a uppercase drive letter resulting in a schema to be registered using a uppercase drive letter but then is later queried using a lowercase drive letter.   
A quick workaround using:
Strawberry perl 5.20.3
Mojolicious version 7.71

    +++ ...\lib\JSON\Validator.pm
    @@ -242,10 +242,12 @@
       $file = path(split '/', $file);
       if (-e $file) {
         $file = $file->realpath;
    +    my $path = $file;
    +    $path =~ s!^(\w:)!lc($1)!e; # Drive letter assumed lower case else where
         warn "[JSON::Validator] Loading schema from file: $file\n" if DEBUG;
    -    return $self->_load_schema_from_text(\$file->slurp), $file;
    -  }
    -  elsif ($file =~ m!^/!) {
    +    return $self->_load_schema_from_text(\$file->slurp), $path;
    +  }
    +  elsif ($file =~ m!^[/|\\]!) {
         warn "[JSON::Validator] Loading schema from URL $url\n" if DEBUG;
         return $self->_load_schema_from_url(Mojo::URL->new($url)->fragment(undef)), "$url";
       }

**Comment by mohawk2** [comment]
I am also seeing `t/bundle.t` hang on Strawberry Perl.

**Comment by jhthorsen** [comment]
@chorny, @vejbaek: Can you see if the change I pushed to master makes any difference?

**Comment by chorny** [comment]
bundle.t no longer hangs and this test passes.

**Comment by jhthorsen** [comment]
Then I hope this is fixed in 2.06.

Thanks! 👍
